src/lib/operators/abstract_aggregate_operator.cpp

```diff
@@ -60,13 +56,19 @@ std::string AbstractAggregateOperator::description(DescriptionMode description_m
 void AbstractAggregateOperator::_validate_aggregates() const {
   const auto input_table = input_table_left();
   for (const auto& aggregate : _aggregates) {
-    if (aggregate.column == INVALID_COLUMN_ID) {
-      Assert(aggregate.function == AggregateFunction::Count, "Aggregate: Asterisk is only valid with COUNT");
+    const auto pqp_column = std::dynamic_pointer_cast<PQPColumnExpression>(aggregate->argument());
+    DebugAssert(pqp_column,
+                "Aggregate operators can currently only handle physical columns, no complicated expressions");
+    const auto column_id = pqp_column->column_id;
+    if (column_id == INVALID_COLUMN_ID) {
+      Assert(aggregate->aggregate_function == AggregateFunction::Count, "Aggregate: Asterisk is only valid with COUNT");
     } else {
-      DebugAssert(aggregate.column < input_table->column_count(), "Aggregate column index out of bounds");
-      Assert(input_table->column_data_type(aggregate.column) != DataType::String ||
-                 (aggregate.function != AggregateFunction::Sum && aggregate.function != AggregateFunction::Avg &&
-                  aggregate.function != AggregateFunction::StandardDeviationSample),
+      DebugAssert(column_id < input_table->column_count(), "Aggregate column index out of bounds");
+      DebugAssert(pqp_column->data_type() == input_table->column_data_type(column_id), "Mismatching column_data_type");
```

Comment:
Can't it happen that we chose a larger numeric type when building sums?

Reply:
This is about the input column, not the output column